File: src/main/java/com/hxiloj/controller/DefaultRestController.java
package com.hxiloj.controller;

import java.security.Principal;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/user")
public class DefaultRestController {
    
	@GetMapping("/principal")
    public Principal user(Principal principal) {
        return principal;
    }
	
}

File: src/main/java/com/hxiloj/controller/DefaultRestController.kt
package com.hxiloj.controller

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import java.security.Principal


@RestController
@RequestMapping("/rest/hello")
class DefaultRestController {
    @GetMapping("/principal")
    fun user(principal: Principal): Principal {
        return principal
    }

    @GetMapping
    fun hello(principal: Principal): String? {
        return "Hello " + principal.name
    }

}